src/Symfony/Component/Validator/Constraints/DefaultPasswordStrengthEstimator.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+final class DefaultPasswordStrengthEstimator
+{
+    /**
+     * @var non-empty-array{score: int, minEntropy: int}
+     */
+    private array $criteria = [
+        ['score' => 4, 'minEntropy' => 120],
+        ['score' => 3, 'minEntropy' => 100],
+        ['score' => 2, 'minEntropy' => 80],
+        ['score' => 1, 'minEntropy' => 60],
+        ['score' => 0, 'minEntropy' => 0],
+    ];
+
+    /**
+     * @param string[] $restrictedData
+     *
+     * @return non-empty-array{score: int, restrictedData: string[]}
```

Comment:
`non-empty-array` does not make sense here, as we return an array shape (which cannot be empty anyway as it requires 2 keys). Use `array{score: int, restrictedData: string[]}`